# Notebook 1: Daten laden

In diesem Schritt laden wir den Datensatz und verschaffen uns einen ersten Überblick über
die Struktur und Inhalte der Daten. Dazu verwenden wir zunächst die Python-Bibliothek **pandas**,
die sich ideal für tabellarische Daten eignet.

In [1]:
import pandas as pd

Dann laden wir die Daten:

In [6]:
df = pd.read_csv('../data/raw/wage_data.csv')
df.head()

,wage,educ,exper,tenure,nonwhite,female,married
0,3.10,11,2,0,0,1,0
1,3.24,12,22,2,0,1,1
2,3.00,11,2,0,0,0,0
3,6.00,8,44,28,0,0,1
4,5.30,12,7,2,0,0,1


Der Datensatz enthält insgesamt **7 Variablen**, die persönliche und berufliche Merkmale von Personen beschreiben:

- **wage** – Stundenlohn (dient als Zielvariable)
- **educ** – Anzahl der Bildungsjahre
- **exper** – Berufserfahrung in Jahren
- **tenure** – Zugehörigkeitsdauer im aktuellen Job
- **nonwhite** – Indikator für ethnische Zugehörigkeit (0 = white, 1 = nonwhite)
- **female** – Geschlecht (0 = männlich, 1 = weiblich)
- **married** – Familienstand (0 = unverheiratet, 1 = verheiratet)

Für das Regressionsmodell definieren wir:

- **Zielvariable (y):** `wage`  
- **Features (X):** `educ`, `exper`, `tenure`, `nonwhite`, `female`, `married`

Damit modellieren wir, wie stark diese sechs Faktoren den Stundenlohn beeinflussen.

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   wage      526 non-null    float64
 1   educ      526 non-null    int64  
 2   exper     526 non-null    int64  
 3   tenure    526 non-null    int64  
 4   nonwhite  526 non-null    int64  
 5   female    526 non-null    int64  
 6   married   526 non-null    int64  
dtypes: float64(1), int64(6)
memory usage: 28.9 KB


### Datenstruktur und Qualität (df.info())

Die `df.info()`-Ausgabe liefert einen Überblick über den Aufbau des Datensatzes:

- **Zeilen:** 526 Beobachtungen  
- **Spalten:** 7 Variablen  
- **Fehlende Werte:** Keine (`526 non-null` in jeder Spalte)  
- **Datentypen:**  
  - `wage` ist eine kontinuierliche numerische Variable (float64)  
  - die übrigen Variablen sind kategoriale oder diskrete numerische Variablen (int64)

Dies bedeutet:
- Der Datensatz ist vollständig und benötigt keine Imputation von fehlenden Werten.
- Alle Variablen sind bereits in einem ML-freundlichen Format.
- Für Modelle wie lineare Regression oder KNN sind keine Typkonvertierungen notwendig.

Die Spalte `wage` kann als **Zielvariable (y)** verwendet werden, während die übrigen 6 Spalten als **Feature-Variablen (X)** geeignet sind.

In [9]:
df.describe()

,wage,educ,exper,tenure,nonwhite,female,married
count,526.000000,526.000000,526.00000,526.000000,526.000000,526.000000,526.000000
mean,5.896103,12.562738,17.01711,5.104563,0.102662,0.479087,0.608365
std,3.693086,2.769022,13.57216,7.224462,0.303805,0.500038,0.488580
min,0.530000,0.000000,1.00000,0.000000,0.000000,0.000000,0.000000
25%,3.330000,12.000000,5.00000,0.000000,0.000000,0.000000,0.000000
50%,4.650000,12.000000,13.50000,2.000000,0.000000,0.000000,1.000000
75%,6.880000,14.000000,26.00000,7.000000,0.000000,1.000000,1.000000
max,24.980000,18.000000,51.00000,44.000000,1.000000,1.000000,1.000000


### Statistische Übersicht der Daten (`df.describe()`)

Die statistische Zusammenfassung gibt einen ersten quantitativen Eindruck über die Verteilung der Variablen im Datensatz. Für jede numerische Spalte werden grundlegende Kennzahlen wie Mittelwert, Standardabweichung sowie Min-/Max-Werte und Quartile angezeigt. 

**Wichtige Beobachtungen:**

- **wage**  
  - Der durchschnittliche Stundenlohn beträgt rund **5.90**.  
  - Der Lohn variiert stark (Standardabweichung ≈ **3.69**), mit Werten zwischen **0.53** und **24.98**.  
  - Diese Streuung deutet auf mögliche Ausreißer hin, die im weiteren Verlauf untersucht werden sollten.

- **educ (Bildungsjahre)**  
  - Median: **12 Jahre**, 75%-Quantil: **14 Jahre**  
  - Der Bereich reicht von **0** bis **18** Jahren, was auf unterschiedliche Bildungsniveaus hinweist.

- **exper (Berufserfahrung)**  
  - Mittelwert: **17 Jahre**, Maximalwert: **51 Jahre**  
  - Die hohe Standardabweichung (≈ **13.57**) zeigt eine große Spannweite zwischen Einsteigern und sehr erfahrenen Beschäftigten.

- **tenure (Betriebszugehörigkeit)**  
  - Median: **4 Jahre**, Maximum **44 Jahre**  
  - Auch hier deutet eine große Streuung auf unterschiedliche Arbeitnehmerprofile hin.

- **nonwhite, female, married**  
  - Dies sind binäre Indikatorvariablen mit Werten 0/1.  
  - Die Mittelwerte entsprechen den relativen Anteilen in der Stichprobe:  
    - Anteil nicht-weißer Personen: **10,2 %**  
    - Anteil Frauen: **47,9 %**  
    - Anteil Verheirateter: **60,8 %**

**Fazit:**  
Die Daten sind vollständig (keine fehlenden Werte) und weisen sowohl kontinuierliche als auch kategoriale Variablen auf. Damit eignen sie sich gut für eine lineare Regression. Die breite Streuung bei Lohn, Erfahrung und Betriebszugehörigkeit kann Einfluss auf die Modellgüte haben und sollte im Modellierungsprozess berücksichtigt werden.

### Visualisierung der Daten

Um ein besseres Verständnis für die Verteilung und Zusammenhänge der Variablen zu bekommen,
visualisieren wir die Daten mit Hilfe von `seaborn` und `matplotlib`. Dies ermöglicht uns:

- Ausreißer zu erkennen  
- Zusammenhänge zwischen Merkmalen zu identifizieren  
- die Verteilung der Zielvariable einzuschätzen  
- potenzielle lineare Beziehungen für die Regression zu überprüfen  